### PR TITLE
[READY] Pretties up chest loot tables

### PR DESCRIPTION
**Pull request by Chuckleberry-Finn**
[why]: This doesn't warrant a change-log cause nothing changes, but the reason as to why is cause I'm probably autistic.

Also:
Sets up a cleaner if()chain-free framework for different probabilities.
Doesn't have a fixed number for rand() that you have to adjust to add/remove things.
Looks nicer?

### CODE REVIEW

**Comment by monkeyboyinc** [comment]
I'm no coding expert, but it looks like this actually does change some of the probabilities. Beforehand, the Resonator Blast and Rapid Repeater Mods both had half the chance of spawning as other loot items, and Unholy water + Inquisitor hardsuit/Lifesteal Mod+Cult Bedsheet were paired together.

I dunno about you, but I'd be pretty pissed to find a bedsheet by itself in a necropolis chest.

**Comment by ChuckTheSheep** [comment]
Oh shoot :x

**Comment by ChuckTheSheep** [comment]
Still handles tiered probability (smaller pools of items after the first). (Just as the original)
Now it can handle bundles of items too. (phew)

File: code/modules/mining/lavaland/necropolis_chests.dm
Comment (ChangelingRain) [inline code/modules/mining/lavaland/necropolis_chests.dm]: this can be an else

File: code/modules/mining/lavaland/necropolis_chests.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mining/lavaland/necropolis_chests.dm]: For consistent formatting's sake, `loot(src)`

File: code/modules/mining/lavaland/necropolis_chests.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mining/lavaland/necropolis_chests.dm]: `i(src)`

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
also I don't think I've ever seen this pick format used that's pretty rad

**Review by ExcessiveUseOfCobblestone — APPROVED** [review]

**Comment by ChuckTheSheep** [comment]
Noticed other instances of similar loot chests, ill work on this to cover the other 2.

**Comment by ChuckTheSheep** [comment]
I noticed abandoned_crates' loot includes a random amount of a choice and I gave up.

Necropolis crates at least have a consistent loot table format, and now share a proc that new()'s off a path or a list of paths.

**Comment by ChuckTheSheep** [comment]
I'm pretty sure that covers all of the giant if()s chains (with no elses) loot chests.

**Comment by ChuckTheSheep** [comment]
I'm pretty sure that covers all of the giant if()s chains (with no elses) loot chests.

**Comment by vuonojenmustaturska** [comment]
You need to rebase, `/obj/item/ore` was moved into `/obj/item/stack/ore`

**Comment by ChuckTheSheep** [comment]
Weird I did it before, desktop must have not pulled properly.

**Comment by vuonojenmustaturska** [comment]
Nah, i merged a shitton of PRs

File: code/modules/mining/abandoned_crates.dm
Comment (vuonojenmustaturska) [inline code/modules/mining/abandoned_crates.dm]: Can you indent this list so that the possible picks are not on the left-most level but rather intended to the level of the pick() function or maybe one more, with sub-lists having one more level of indentation?

File: code/modules/mining/abandoned_crates.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mining/abandoned_crates.dm]: outdated comment

Comment (ChuckTheSheep) [reply code/modules/mining/abandoned_crates.dm]: Mentioning necrocrates is cause they share that proc, I just didn't know if I should stick the proc in crates.dm or leave it near a more relevant code.

File: code/modules/mining/abandoned_crates.dm
Comment (ChuckTheSheep) [reply code/modules/mining/abandoned_crates.dm]: Ill try again later today, it was giving me trouble before.

**Comment by ChuckTheSheep** [comment]
Mobile github ugh...

File: code/modules/mining/abandoned_crates.dm
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mining/abandoned_crates.dm]: throw it in crates.dm and have the comment say something like `// used in abandoned crates and necro chests`

File: code/modules/mining/abandoned_crates.dm
Comment (ChuckTheSheep) [reply code/modules/mining/abandoned_crates.dm]: Done and done.

**Comment by optimumtact** [comment]
I don't know it seems worse to me, I'll leave this to another maintainer @tgstation/commit-access

**Comment by Cyberboss** [comment]
That's still doing something for those who prefer to, while not forcing
those who don't to participate.

**Comment by Cyberboss** [comment]
Btw, nothing is interactable aside from the computers (including other
players)

**Comment by Cyberboss** [comment]
Fuck I replied to the wrong thread

**Comment by lzimann** [comment]
yea i think this looks worse tbh

**Comment by ChuckTheSheep** [comment]
To each their own. ¯\_( :'( )_/¯

**Comment by ChuckTheSheep** [comment]
It comes down to preference of course. 
The only thing this would help with is not having open if()s (which aren't that big of a deal) and making editing not that tedious like here: https://github.com/tgstation/tgstation/pull/35364/files  but even then it's not that's also not that big of a deal.

File: code/game/objects/structures/crates_lockers/crates.dm
Comment (ShizCalev) [inline code/game/objects/structures/crates_lockers/crates.dm]:
```
/obj/structure/closet/crate/proc/pathorlist_to_loot(loot)
	if(!loot)
		return
	if(ispath(loot))
		new loot(src)
		return
	if(LAZYLEN(loot))
		for(var/i in loot)
			if(!ispath(i))
				continue
			for(var/ii in 1 to (loot[i] ? loot[i] : 1))//does i-in-loot have an associated value? if so use it, otherwise 1
				new i(src)
```

File: code/modules/mining/abandoned_crates.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mining/abandoned_crates.dm]: Just to be clear, these comments are only true because the total relative values add up to be 100

Any additional items added will make all related comments incorrect.

**Comment by ExcessiveUseOfCobblestone** [comment]
You think the V;P format is less appealing than ifto-pacolypse?

**Comment by optimumtact** [comment]
yes

**Comment by Qustinnus** [comment]
why dont we just use pickweight for this?

**Comment by ChuckTheSheep** [comment]
This uses the internal weight system with pick(). The chests have really specific cases where a general proc that gets loaded with a loot-table wouldn't be easy to maintain. But I was made aware this could be done better as loot-spawners/landmarks. So I'll just close it.

**Pull request closed without merging**
